host/frontend/host-orchestrator/instancemanager_test.go

```diff
@@ -55,3 +64,158 @@ func TestCreateCVDInvalidRequests(t *testing.T) {
 		}
 	}
 }
+
+type FakeFetchCVDDownloader struct {
+	t       *testing.T
+	content string
+}
+
+func (d *FakeFetchCVDDownloader) Download(dst io.Writer, buildID string, accessToken string) error {
+	r := strings.NewReader(d.content)
+	if _, err := io.Copy(dst, r); err != nil {
+		d.t.Fatal(err)
+	}
+	return nil
+}
+
+func TestFetchCVDHandler(t *testing.T) {
+	dir := t.TempDir()
+	if err := os.WriteFile(filepath.Join(dir, "fetch_cvd_1"), []byte(string("000")), 0644); err != nil {
+		t.Fatal(err)
+	}
+	downloader := &FakeFetchCVDDownloader{
+		t:       t,
+		content: "111",
+	}
+
+	t.Run("binary is not downloaded as it already exists", func(t *testing.T) {
+		h := NewFetchCVDHandler(dir, downloader)
+
+		err := h.Download("1", "tokenAbCDe")
+
+		if err != nil {
+			t.Errorf("epected <<nil>> error, got %#v", err)
+		}
+		content, err := ioutil.ReadFile(filepath.Join(dir, "fetch_cvd_1"))
+		if err != nil {
+			t.Fatal(err)
+		}
+		actual := string(content)
+		expected := "000"
+		if actual != expected {
+			t.Errorf("expected <<%q>>, got %q", expected, actual)
+		}
+	})
+
+	t.Run("binary is downloaded", func(t *testing.T) {
+		h := NewFetchCVDHandler(dir, downloader)
+
+		h.Download("2", "tokenAbCDe")
+
+		content, _ := ioutil.ReadFile(filepath.Join(dir, "fetch_cvd_2"))
+		actual := string(content)
+		expected := "111"
+		if actual != expected {
+			t.Errorf("expected <<%q>>, got %q", expected, actual)
+		}
+	})
+
+	if err := os.Remove(filepath.Join(dir, "fetch_cvd_1")); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.Remove(filepath.Join(dir, "fetch_cvd_2")); err != nil {
+		t.Fatal(err)
+	}
+}
+
+type AlwaysFailsFetchCVDDownloader struct{}
+
+func (d *AlwaysFailsFetchCVDDownloader) Download(dst io.Writer, buildID string, accessToken string) error {
+	return fmt.Errorf("downloading failed")
+}
+
+func TestFetchCVDHandlerDownloadingFails(t *testing.T) {
+	dir := t.TempDir()
+	h := NewFetchCVDHandler(dir, &AlwaysFailsFetchCVDDownloader{})
+
+	err := h.Download("1", "tokenAbCDe")
+
+	if err == nil {
+		t.Errorf("expected an error")
+	}
+	if _, err := os.Stat(filepath.Join(dir, "fetch_cvd_1")); err == nil {
+		t.Errorf("file must not have been created")
+	}
+}
+
+func TestABFetchCVDDownloaderDownload(t *testing.T) {
+	var req *http.Request
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		req = r
+		w.Write([]byte("OK"))
+	}))
+	defer ts.Close()
+
+	t.Run("request uri", func(t *testing.T) {
```

Comment:
TBH I don't see any value in this test other than checking the strings. IMO it's testing code/impl instead of contract, which means changing the implementation without changing the semantics is going to require changing the test too.
Of course, that's subjective, so you can leave it as is and deal with it if/when it breaks for the wrong reason.